backend/app/core/middleware.py

```
@@ -0,0 +1,281 @@
+            # Don't log sensitive data
+            if body_str and any(field in body_str.lower() for field in ['password', 'token', 'key']):
+                body_str = "[REDACTED]"
+            
+            logger.info(
+                f"Incoming request - Method: {request.method}, "
+                f"URL: {request.url}, "
+                f"Headers: {dict(request.headers)}, "
+                f"Body: {body_str}, "
+                f"Correlation-ID: {correlation_id}"
+            )
+        except Exception as e:
```

Ruff (0.14.2)

Do not catch blind exception: `Exception`

(BLE001)